R/tools.R

```diff
@@ -26,6 +26,45 @@ lint_dir <- function(path) {
   lints
 }
 
+lint_file <- function(path) {
+  if (interactive()) {
+    message(cli::format_inline("Linting {.file {path}}"))
+  }
+  lints <- lintr::lint(path)
+  # Use the actual path provided by the user (typically relative) to make results more readable.
+  # (`lintr::lint()` normalizes paths, e.g. `app/main.R` becomes a full absolute path.)
+  for (i in seq_along(lints)) {
+    lints[[i]]$filename <- path
+  }
+  lints
+}
+
+lint_path <- function(path) {
+  # Check if path is a directory or a file such that nothing happens when the path is invalid.
+  if (fs::is_dir(path)) {
+    lint_dir(path)
+  } else if (fs::is_file(path)) {
+    lint_file(path)
+  } else {
+    cli::cli_abort("Unexpected invalid path: {.file {path}}.")
+  }
+}
+
+check_paths <- function(paths) {
+  readable <- fs::file_access(paths, mode = "read")
+
+  if (any(!readable))
+    cli::cli_abort(
+      c(
+        "Cannot lint an invalid path.",
+        i = "Please check that {cli::col_blue('paths')} contain only valid paths.",
+        i = "The following path{?s} cannot be read: {.file {paths[!readable]}}."
+      )
+    )
+
+  paths
```

Comment:
1. Use braces for readability.
2. Use `arg` class for `paths` in the CLI message. See [inline markup](https://cli.r-lib.org/reference/inline-markup.html#classes) for reference.
3. Pass `call = NULL` to avoid printing the function name. We are reporting an error which is fixable by the user; the name of this internal helper function won't help them.
4. No need to return `paths` - they are not modified in any way now.

```suggestion
  if (any(!readable)) {
    cli::cli_abort(
      c(
        "Cannot lint an invalid path.",
        i = "Please check that {.arg paths} contain only valid paths.",
        i = "The following path{?s} cannot be read: {.file {paths[!readable]}}."
      ),
      call = NULL
    )
  }
```
```